sdk/core/azure-core-experimental/src/main/java/com/azure/core/experimental/models/TimeInterval.java

```diff
@@ -0,0 +1,264 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.core.experimental.models;
+
+import com.azure.core.annotation.Immutable;
+import com.azure.core.util.logging.ClientLogger;
+
+import java.time.Duration;
+import java.time.OffsetDateTime;
+import java.util.Objects;
+
+/**
+ * Class to represent a time interval.
+ */
+@Immutable
+public final class TimeInterval {
+    public static final TimeInterval ALL = new TimeInterval(OffsetDateTime.MIN, OffsetDateTime.MAX);
+
+    public static final TimeInterval LAST_5_MINUTES = new TimeInterval(Duration.ofMinutes(5));
+    public static final TimeInterval LAST_30_MINUTES = new TimeInterval(Duration.ofMinutes(30));
+
+    public static final TimeInterval LAST_1_HOUR = new TimeInterval(Duration.ofHours(1));
+    public static final TimeInterval LAST_4_HOURS = new TimeInterval(Duration.ofHours(4));
+    public static final TimeInterval LAST_12_HOURS = new TimeInterval(Duration.ofHours(12));
+
+    public static final TimeInterval LAST_DAY = new TimeInterval(Duration.ofDays(1));
+    public static final TimeInterval LAST_2_DAYS = new TimeInterval(Duration.ofDays(2));
+    public static final TimeInterval LAST_3_DAYS = new TimeInterval(Duration.ofDays(3));
+    public static final TimeInterval LAST_7_DAYS = new TimeInterval(Duration.ofDays(7));
```

Comment:
Do we need to have these constants?

Reply:
We don't require these constants but it serves as a convenience. All these constants are time intervals we have in Azure Portal for metrics dashboard.